azurerm/internal/services/compute/managed_disk_resource_test.go

```diff
@@ -322,6 +323,147 @@ func TestAccManagedDisk_attachedStorageTypeUpdate(t *testing.T) {
+			Check: resource.ComposeTestCheckFunc(
+				testCheckManagedDiskExists(data.ResourceName, true),
+				resource.TestCheckResourceAttr(data.ResourceName, "network_access_policy", "DenyAll"),
+			),
+		},
+	})
+}
+
+func TestAccAzureRMManagedDisk_import_withNetworkPolicy(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_managed_disk", "test")
+	r := ManagedDiskResource{}
+
+	data.ResourceTest(t, r, []resource.TestStep{
+		{
+			Config: testAccAzureRMManagedDisk_create_withNetworkPolicy(data),
+			Check: resource.ComposeTestCheckFunc(
+				testCheckManagedDiskExists(data.ResourceName, true),
+			),
+		},
+		{
+			Config:      testAccAzureRMManagedDisk_import_withNetworkPolicy(data),
+			ExpectError: acceptance.RequiresImportError("azurerm_managed_disk"),
+		},
+	})
+}
+
+func TestAccAzureRMManagedDisk_create_withAllowPrivateNetworkPolicy(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_managed_disk", "test")
+	r := ManagedDiskResource{}
+
+	data.ResourceTest(t, r, []resource.TestStep{
+		{
+			Config: testAccAzureRMManagedDisk_create_withAllowPrivateNetworkPolicy(data),
+			Check: resource.ComposeTestCheckFunc(
+				testCheckManagedDiskExists(data.ResourceName, true),
+			),
+		},
+	})
+}
+
+func TestAccAzureRMManagedDisk_update_withAllowPrivateNetworkPolicy(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_managed_disk", "test")
+	r := ManagedDiskResource{}
+
+	data.ResourceTest(t, r, []resource.TestStep{
+		{
+			Config: testAccAzureRMManagedDisk_create_withAllowPrivateNetworkPolicy(data),
+			Check: resource.ComposeTestCheckFunc(
+				testCheckManagedDiskExists(data.ResourceName, true),
+				resource.TestCheckResourceAttr(data.ResourceName, "network_access_policy", "AllowPrivate"),
+			),
+		},
+		{
+			Config: testAccAzureRMManagedDisk_update_withAllowPrivateNetworkPolicy(data),
+			Check: resource.ComposeTestCheckFunc(
+				testCheckManagedDiskExists(data.ResourceName, true),
+				resource.TestCheckResourceAttr(data.ResourceName, "network_access_policy", "AllowPrivate"),
+			),
+		},
+	})
+}
+
+func TestAccAzureRMManagedDisk_import_withAllowPrivateNetworkPolicy(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_managed_disk", "test")
+	r := ManagedDiskResource{}
+
+	data.ResourceTest(t, r, []resource.TestStep{
+		{
+			Config: testAccAzureRMManagedDisk_create_withAllowPrivateNetworkPolicy(data),
+			Check: resource.ComposeTestCheckFunc(
+				testCheckManagedDiskExists(data.ResourceName, true),
+			),
+		},
+		{
+			Config:      testAccAzureRMManagedDisk_import_withAllowPrivateNetworkPolicy(data),
+			ExpectError: acceptance.RequiresImportError("azurerm_managed_disk"),
+		},
+	})
+}
+
+// nolint unparam
+func testCheckManagedDiskExists(resourceName string, shouldExist bool) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		client := acceptance.AzureProvider.Meta().(*clients.Client).Compute.DisksClient
+		ctx := acceptance.AzureProvider.Meta().(*clients.Client).StopContext
```

Comment:
could we remove this and use `data.CheckWithClient`? as we are removing all usages of acceptance.AzureProvider for binary testing